hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/operators/special/HazelcastUDTObjectToJsonFunction.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.validate.operators.special;
+
+import com.hazelcast.jet.sql.impl.validate.HazelcastCallBinding;
+import com.hazelcast.jet.sql.impl.validate.operators.common.HazelcastFunction;
+import com.hazelcast.jet.sql.impl.validate.operators.typeinference.ReplaceUnknownOperandTypeInference;
+import com.hazelcast.jet.sql.impl.validate.types.HazelcastJsonType;
+import com.hazelcast.jet.sql.impl.validate.types.HazelcastObjectType;
+import com.hazelcast.jet.sql.impl.validate.types.HazelcastObjectTypeReference;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlOperandCountRange;
+import org.apache.calcite.sql.type.SqlOperandCountRanges;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+public class HazelcastUDTObjectToJsonFunction extends HazelcastFunction {
```

Comment:
This is a bit problematic. OBJECT is used for dynamic params ("untyped" OBJECT that is) and technically there is no difference between `CAST(? AS JSON)` and `CAST(this AS JSON)` in terms of types involved. Its probably possible to disable CAST(OBJECT AS JSON) for cases where input OBJECT is not a dynamic param, but I suspect it would be complicated to implement.